test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/SignerAndEndpointOverridesTest.java

```diff
@@ -0,0 +1,228 @@
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static software.amazon.awssdk.core.client.config.SdkAdvancedClientOption.SIGNER;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.auth.signer.AwsSignerExecutionAttribute;
+import software.amazon.awssdk.awscore.endpoints.AwsEndpointAttribute;
+import software.amazon.awssdk.awscore.endpoints.authscheme.SigV4AuthScheme;
+import software.amazon.awssdk.awscore.endpoints.authscheme.SigV4aAuthScheme;
+import software.amazon.awssdk.core.async.AsyncRequestBody;
+import software.amazon.awssdk.core.signer.Signer;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.endpoints.Endpoint;
+import software.amazon.awssdk.http.SdkHttpFullRequest;
+import software.amazon.awssdk.http.auth.aws.signer.AwsV4FamilyHttpSigner;
+import software.amazon.awssdk.http.auth.aws.signer.AwsV4HttpSigner;
+import software.amazon.awssdk.http.auth.aws.signer.AwsV4aHttpSigner;
+import software.amazon.awssdk.http.auth.aws.signer.RegionSet;
+import software.amazon.awssdk.http.auth.spi.scheme.AuthSchemeOption;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.protocolquery.ProtocolQueryAsyncClient;
+import software.amazon.awssdk.services.protocolquery.ProtocolQueryClient;
+import software.amazon.awssdk.services.protocolquery.auth.scheme.ProtocolQueryAuthSchemeProvider;
+import software.amazon.awssdk.services.protocolquery.endpoints.ProtocolQueryEndpointProvider;
+
+/**
+ * Tests to ensure that parameters set when endpoint and auth-scheme resolution occurs get propagated to the overriden
+ * signer (i.e. pre-SRA signers).
+ */
+public class SignerAndEndpointOverridesTest {
+
+    public Signer mockSigner = mock(Signer.class);
+
+    @Test
+    public void test_whenV4EndpointAuthSchemeWithSignerOverride_thenEndpointParamsShouldPropagateToSigner() {
+        ProtocolQueryClient client = ProtocolQueryClient
+            .builder()
+            .authSchemeProvider(v4AuthSchemeProviderOverride())
+            .credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.create("akid", "skid")))
+            .endpointProvider(v4EndpointProviderOverride())
+            .region(Region.US_WEST_2)
+            .overrideConfiguration(o -> o.putAdvancedOption(SIGNER, mockSigner))
+            .build();
+
+        try {
+            client.streamingInputOperation(r -> {}, RequestBody.fromString("test"));
+        } catch (Exception expected) {
+        }
```

Comment:
Can we switch instead to use an interceptor and assert that the exception is the expected one?, See for instance the setup in the [tests here](https://github.com/aws/aws-sdk-java-v2/blob/dbc7909b36acc3e92595f7e01e1d1cdd37da92b6/test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/OverrideConfigurationPluginsTest.java#L38). Rationale, this ensures that the exception thrown is exactly what we expect instead of something else failing in the middle.

For instance, the assertion will be instead

```
    assertThatThrownBy(() -> client.streamingInputOperation(r -> { }, RequestBody.fromString("test")))
            .hasMessageContaining("boom!");
```

Reply:
Point taken. We don't need the interceptor, we can just use the mocked signer to do that.